src/change_list/mod.rs

```diff
@@ -235,18 +235,18 @@ impl ChangeListBuilder<'_> {
 
     pub fn set_attribute(&mut self, name: &str, value: &str) {
         debug_assert!(self.traversal_is_committed());
-        if name == "class" {
-            let class_id = self.ensure_string(value);
-            debug!("emit: set_class({:?})", value);
-            self.state.emitter.set_class(class_id.into());
-        } else {
+        // if name == "class" {
+        //     let class_id = self.ensure_string(value);
+        //     debug!("emit: set_class({:?})", value);
+        //     self.state.emitter.set_class(class_id.into());
+        // } else {
```

Comment:
Using `className` when it is valid to do so is a significant speedup.

I haven't read closely enough to the original issue yet to have a strong grasp on what the ways that SVG and class name setting were interacting poorly are, but I think the best fix is probably to only use the fast path when `name == "class" && namespace.is_none()`. This will require a little more plumbing to pass the namespace in, if any, but shouldn't be too difficult.